### PR TITLE
Backend add some devtools

### PR DESCRIPTION
**Pull request by stegben**
Add some tools for better development experiences.
- [x] gitignore
- [x] editorconfig
- [x] linter and auto-formatting (will try [autopep8](), [yapf](https://github.com/google/yapf) and [black](https://github.com/psf/black))
edit: use `black` for simplicity and its speed.
- [x] Makefile
- [x] use `dotenv` for configurations.

related: #3

### CODE REVIEW

**Comment by stegben** [comment]
commit 先散著打，之後再 squash

**Comment by Yukaii** [comment]
不用 squash 也沒關係啦，commit 爆 🤣

File: backend/gis_project/settings.py [hunk 1]
```diff
@@ -31,63 +31,61 @@
...
-        'PASSWORD': 'postgres',
-        'HOST': 'db',
-        'PORT': 5432
+    "default": {
+        "ENGINE": "django.db.backends.postgresql",
+        "NAME": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_NAME", "postgres"),
+        "USER": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_USER", "postgres"),
+        "PASSWORD": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_PASSWORD", "postgres"),,
+        "HOST": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_HOST", "db"),,
+        "PORT": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_PORT", "5423"),,
```
Comment (Yukaii) [inline backend/gis_project/settings.py]: ,,

File: backend/gis_project/settings.py [hunk 2]
```diff
@@ -31,63 +31,61 @@
...
-        'NAME': 'postgres',
-        'USER': 'postgres',
-        'PASSWORD': 'postgres',
-        'HOST': 'db',
-        'PORT': 5432
+    "default": {
+        "ENGINE": "django.db.backends.postgresql",
+        "NAME": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_NAME", "postgres"),
+        "USER": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_USER", "postgres"),
+        "PASSWORD": os.environ.get("DISFACTORY_BACKEND_DEFAULT_DB_PASSWORD", "postgres"),,
```
Comment (Yukaii) [inline backend/gis_project/settings.py]: ,,

**Comment by Yukaii** [comment]
發現 docker-compose up 起不來 #10

File: backend/gis_project/settings.py (hunk 1 above)
Comment (stegben) [reply backend/gis_project/settings.py]: fixed

File: backend/gis_project/settings.py (hunk 2 above)
Comment (stegben) [reply backend/gis_project/settings.py]: fixed

**Comment by stegben** [comment]
我這邊可以 `docker-compose up`:
```
Creating backend_db_1 ... done
Creating backend_web_1 ... done
Attaching to backend_db_1, backend_web_1
db_1   | 2019-09-29 07:13:57.717 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
db_1   | 2019-09-29 07:13:57.717 UTC [1] LOG:  listening on IPv6 address "::", port 5432
db_1   | 2019-09-29 07:13:57.721 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
db_1   | 2019-09-29 07:13:57.771 UTC [24] LOG:  database system was shut down at 2019-09-29 07:13:48 UTC
db_1   | 2019-09-29 07:13:57.778 UTC [1] LOG:  database system is ready to accept connections
web_1  | Watching for file changes with StatReloader
web_1  | Performing system checks...
web_1  | 
web_1  | System check identified no issues (0 silenced).
web_1  | 
web_1  | You have 18 unapplied migration(s). Your project may not work properly until you apply the migrations for app(s): admin, auth, contenttypes, sessions, users.
web_1  | Run 'python manage.py migrate' to apply them.
web_1  | September 29, 2019 - 07:13:59
web_1  | Django version 2.2.3, using settings 'gis_project.settings'
web_1  | Starting development server at http://0.0.0.0:8000/
web_1  | Quit the server with CONTROL-C.
```

**Comment by stegben** [comment]
@Yukaii 這樣應該就沒問題了

**Comment by stegben** [comment]
看起來沒啥問題就先 merge 了

**Pull request merged**